### PR TITLE
Add tests for DnsDiscovery and BtcFixedFormat

### PR DESCRIPTION
**Pull request by MethDamon**

### CODE REVIEW

**Comment by MethDamon** [comment]
@schildbach Could you take a look at this? I saw that the DnsDiscovery class has 0% test coverage so I figured I'd write a test for it

File: core/src/test/java/org/bitcoinj/net/discovery/DnsDiscoveryTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DnsDiscoveryTest {
+
+    @Test
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/net/discovery/DnsDiscoveryTest.java]: These tests probably need the network. I'm happy to merge them, but they need to be excluded from regular build. See build.gradle for that.

File: core/src/test/java/org/bitcoinj/net/discovery/DnsDiscoveryTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        InetSocketAddress[] inetSocketAddresses = dnsSeedDiscovery.getPeers(0, 100, TimeUnit.MILLISECONDS);
+        assertNotEquals(0, inetSocketAddresses.length);
+    }
+
+    @Test(expected = PeerDiscoveryException.class)
+    public void testGetPeersThrowsPeerDiscoveryExceptionForUnknownHost() throws PeerDiscoveryException {
+        DnsDiscovery.DnsSeedDiscovery dnsSeedDiscovery = new DnsDiscovery.DnsSeedDiscovery(MainNetParams.get(), "unknown host");
+        dnsSeedDiscovery.getPeers(0, 100, TimeUnit.MILLISECONDS);
+    }
+
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/net/discovery/DnsDiscoveryTest.java]: Can you remove these extra blank lines and make sure there is a proper line ending at EOF?

File: core/src/test/java/org/bitcoinj/utils/BtcFixedFormatTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    @Before
+    public void initBtcFixedFormat() {
+        this.btcFixedFormat = new BtcFixedFormat(Locale.getDefault(), input, 2, groups);
+    }
+
+    @Test
+    public void testPrefixLabel() {
+        assertEquals(String.format("%s%s %s", expectedOutput, "format", btcFixedFormat.pattern()), btcFixedFormat.toString());
+    }
+
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/utils/BtcFixedFormatTest.java]: Same here.

File: core/src/test/java/org/bitcoinj/utils/BtcFixedFormatThrowsIllegalStateExceptionTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.Locale;
+
+public class BtcFixedFormatThrowsIllegalStateExceptionTest {
+
+    @Test(expected = IllegalStateException.class)
+    public void testScaleThrowsIllegalStateException() {
+        BtcFixedFormat btcFixedFormat = new BtcFixedFormat(Locale.getDefault(), BtcFormat.MILLICOIN_SCALE, 2, new ArrayList<Integer>());
+        btcFixedFormat.scale(new BigInteger("2000"), 0);
+    }
+
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/utils/BtcFixedFormatThrowsIllegalStateExceptionTest.java]: And here.

**Comment by schildbach** [comment]
Please squash any fixes into the respective commit. In the end, two commits seem appropriate: One for DnsDiscoveryTest, one for BtcFixedFormatTest.

**Comment by MethDamon** [comment]
@schildbach Fixed according to your comments. Would be happy if you could take another look at this :)

**Comment by schildbach** [comment]
I removed one seed from DnsDiscoveryTest.testBuildDiscoveries(), because it didn't resolve for me. Then I merged. Thanks!

**Pull request closed without merging**